test/k8sT/Services.go

```diff
@@ -1642,14 +1642,27 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 
 			terminateServiceEndpointPod()
 
+			By("Checking if client pod terminated gracefully")
+			ctx, cancel = context.WithCancel(context.Background())
+			res = kubectl.LogsStream(helpers.DefaultNamespace, clientPod, ctx)
+			// The log message indicates that the connectivity between client and
+			// server was intact even after the service endpoint pod was terminated,
+			// and that the client connection terminated gracefully.
+			Expect(res.WaitUntilMatch("exiting on graceful termination")).To(BeNil(),
+				"%s is not in the output after timeout", res.GetStdOut())
```

Comment:
Typically, `Eventually` should be used over `WaitUntilMatch`. I imagine the latter was added due to lack of knowledge of the former.

Reply:
>// WaitUntilMatch waits until the given substring is present in the `CmdRes.stdout`
>// If the timeout is reached it will return an error.

The helper is a special case of `Eventually` (although it uses a different function than `Eventually` underneath) as it operates on `CmdRes*`. If we were to use eventually, we'll need to replicate the parts of `WaitUntilMatch` anyway?

Reply:
AFAIU, `Eventually` could be used with just:
```go
Eventually(func() bool {
    return strings.Contains(res.OutputPrettyPrint(), "exiting on graceful termination")
}, helpers.HelperTimeout, time.Second).Should(BeTrue(), "%s is not in the output after timeout", res.GetStdOut())
```
That said, I'm not clear why `Eventually` would be preferred here. @christarazi What's the reason?

Reply:
The reason is so that we can remove cruft from our Ginkgo test suite :slightly_smiling_face: